readme.txt
=======================================================
    Genmod

    Version 1.0
    Copyright 2005 - 2007 Genmod Development team

    # $Id: readme.txt 29 2022-07-17 13:18:20Z Boudewijn $
=======================================================

CONTENTS
     1.  LICENSE
     2.  INTRODUCTION
     3.  SYSTEM REQUIREMENTS
     4.  PHP SETTINGS
     5.  QUICK INSTALL
     6.  INSTALLATION
     7.  UPDATING GEDCOMS
     8.  THEMES
     9.  MULTIMEDIA OBJECTS
    10.  RSS FEED
    11.  SECURITY
    12.  LANGUAGES
    13.  NON-STANDARD GEDCOM CODES
    14.  LANGUAGE EXTENSION FILES
    15.  BACKUP
    16.  RESTORE
    17.  ENTERPRISE VERSION

-------------------------------------------------------
LICENSE

Genmod: Genealogy Editor and Viewer
Copyright (C) 2005 - 2007  Genmod Development team

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 2 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

See the file GPL.txt included with this software for more
detailed licensing information.

-------------------------------------------------------
INTRODUCTION

Genmod is a genealogy program which allows you to view 
and edit your genealogy on your website.  Genmod has full editing 
capabilities, full privacy functions, and supports multimedia like photos 
and document images.  Genmod also simplifies the process of 
collaborating with others working on your family lines.  Your latest 
information is always on your web site and available for others to see.  
For more information visit http://genmod.sourceforge.net/.

Genmod is taking after the works by John Finaly on the PhpGedView project.

Genmod is Open Source software that has been produced by people from 
many countries freely donating their time and talents to the project.  All 
service, support, and future development is dependent on the time 
developers are willing to donate to the project, often at the expense of 
work, recreation, and family.  Beyond the few donations received from 
users, developers receive no compensation for the time they spend working 
on the project.  There is also no outside source of revenue to support the 
project.

-------------------------------------------------------
SYSTEM REQUIREMENTS

Genmod requires the following:
* A web server (Apache, IIS)
* PHP 4.3.0 or higher
* MySQL 4.1 or higher
* Enough diskspace (around 20MB)

The default installations of PHP on most servers should provide
you with all of the PHP functionality you should need.

To use the reporting engine, PHP needs to be compiled with XML support.  
This is compiled into PHP by default unless it is specifically disabled.  
See http://us3.php.net/manual/en/ref.xml.php

Some features of Genmod require the GD library and that PHP be compiled
with GD support.  Most precompiled versions of PHP include GD support.  If 
you are compiling PHP yourself you will need to configure it with the 
    --with-gd 
option.  See http://us3.php.net/manual/en/ref.image.php

The advanced calendar features of Genmod for converting Gregorian dates 
to the Hebrew or Jewish calendars require that PHP be compiled with the
    --enable-calendar 
configuration option.  See http://us2.php.net/manual/en/ref.calendar.php 
for more information.

For GEDCOMs larger than 2MB, you will likely need to research different
hosting options and find one that will accept a request to modify the
default memory and time limits built into PHP.  See the FAQ at
http://www.sourceforge.net/projects/genmod for more information about large GEDCOMs.

-------------------------------------------------------
PHP SETTINGS

When you experience trouble with logging into the program or nothing is being
displayed take a look at your PHP settings. Here are some important settings
with their recommendend values.

* output_buffering = 4096
* allow_call_time_pass_reference = Off
* register_globals = Off
* register_argc_argv = Off
* magic_quotes_gpc = Off
* session.auto_start = 0
-------------------------------------------------------
QUICK INSTALLATION

Follow the instructions in this section to install Genmod if you are
already familiar with the program or are familiar with installing other PHP 
web applications.

 1.  Upload the files to your web server.
 2.  Set Write permissions on config.php and the "index" directory.  For 
     optimal security, you should move the "index" directory to a location 
     where it is not accessible from the Internet.
 3.  Point your browser to the directory where you uploaded your Genmod 
     files (for example, http://www.yourserver.com/Genmod/).
 4.  Enter your configuration settings.  If you moved the index directory, 
     be sure to specify the correct location to it on this page.  Save the 
     configuration parameters.
 5.  You are taken through the installation process.
 6.  Login as this user and upload your GEDCOM file.
 7.  Save the GEDCOM configuration settings.
 8.  Import the GEDCOM.

Optional Steps
 9.  If you want to use the language editing features you will need to set 
     Write permissions for all of the files in the ./languages folder.
10.  If you want to upload media files using the Upload Media section of 
     the Admin menu then you need to set Write permissions for your ./media 
     and ./media/thumbs directories.
11.  If you want to edit your GEDCOM file online, the GEDCOM file must have
     Write permissions set for the PHP user.
12.  If you want to use the Backup feature of the Upgrade utility in 
     Genmod you will need to either set Write permission on the 
     Genmod folder itself or create a folder named "backup" with Write 
     permissions.  Write permissions for the Genmod folder can be 
     removed as soon as the backup folder is in place and has the 
     appropriate permissions.
13.  For security you should set the permissions back to Read-only when you 
     are done editing or uploading files.

-------------------------------------------------------
INSTALLATION

Follow these instructions if you are not familiar with Genmod or 
installing PHP applications.

*A.  Upload Program Files:
To install Genmod, unzip the compressed package and upload the files to 
a directory on your web server.  If you have limited space on your server, 
you can save space in the following ways:
1.  Delete the themes from the themes folder that you do not plan to use.
2.  Delete some or all of the language files that you do not want. English 
    files are named configure_help.en.txt, countries.en.txt, facts.en.txt, 
    help_text.en.txt and lang.en.txt.  French files, for example, are named 
    with ".fr." in place of ".en.".  Hebrew files use ".he." in place of 
    ".en.", and so on. All active languages are loaded into the database,
    therefore they can be removed. If you need to restore any of the languages
    make sure the files are back in the languages folder.
    The English language files should not be deleted.  They are always loaded 
    before the files for the selected language are loaded.  This ensures 
    that all language variables are defined, and that the English version 
    will be used when a given variable is missing in the new language.
3.  Do not upload the "places" folder.  This folder contains maps for some
    countries.  It also contains text files containing state, county, and 
    place names.  Its purpose is to allow you to enter place names by 
    picking them from lists.    

For optimal security, you may want to move the "index" directory to a 
different location outside of your Internet accessible space.  You will 
specify the location of this directory during the online configuration.  
See the SECURITY section for more information.

*B.  Required File Permissions:
Genmod requires that Read permissions be set for all files in the
Genmod directory tree.  Some hosts also require Execute permissions
(chmod 755).  Genmod requires full Write permissions on the index 
directory (chmod 777 under most hosting configurations).  Genmod also 
requires that Write permissions (chmod 777) be set temporarily for the 
config.php file.

To help with the setting of permissions a file called setpermissions.php 
has been included with the project.  This file will attempt to set 777 
permissions to the config.php, ./index, and all of the files inside the 
./index directory.  Because host settings vary on the ability of PHP 
programs to set file permissions, you have to run this file manually.

If at any time you have trouble during configuration, check your 
permissions again.

There are some advanced features that require more Write permissions to be 
set.  If you want to use the language editing features you will need to set 
Write permissions for all of the files in the ./languages folder (chmod 
777).  If you want to upload media files using the Upload Media section of 
the Admin menu then you need to set Write permissions (chmod 777) for your 
./media and ./media/thumbs directories.  If you want to edit your GEDCOM 
file online, the GEDCOM file must have Write permissions set to the PHP 
user (chmod 777).

*C.  Configuration:
Next point your web browser to the Genmod folder
(for example, http://www.yourserver.com/Genmod/) to automatically 
begin the online configuration process.  Information about each of the 
configuration options can be found online by clicking on the question mark 
(?) near each label.

You may reconfigure Genmod at any time by taken the following steps:
1. Log in as an administrator
2. Go to the admin page
3. Click on "maintenance"
4. Click on "Restart setup"

*D.  Add GEDCOM file
To add GEDCOM files to the system, you can upload your GEDCOM file using 
the "Upload GEDCOM" option from the Admin menu.  All files uploaded using 
the "Upload GEDCOM" page are saved in your index directory.  You can also 
upload your GEDCOM manually using FTP or any other file upload method.  
Most hosts limit the size of files that can be uploaded from a web form for 
security reasons, so you may be forced to use a manual method.  You may 
also upload your GEDCOM in ZIP format, either manually or using the 
"Upload GEDCOM" option.  Make sure to enter the filename of the ZIP file.  
Genmod will automatically unpack the ZIP file and use the GEDCOM file 
inside it.  Be sure to create the ZIP file to contain only one GEDCOM file.

*E.  Set GEDCOM Configuration Settings
After uploading your GEDCOM, you will be asked to set the configuration
parameters for it.  There are too many parameters to list all of their 
options in this document.  Please use the online Help documentation to 
guide you through the configuration process.

*F.  Validate GEDCOM
After you save the GEDCOM configuration Genmod will validate your 
GEDCOM and automatically fix any errors that it can.  If any errors found 
in the GEDCOM require user input, you will be prompted to choose how to 
proceed.  Again use the online Help ? for more information.

*G.  Import GEDCOM
You are almost done.  This is the final step before you can begin viewing 
your data.  After validating the GEDCOM and fixing any errors, you will 
need to import the GEDCOM into the data store.  During the Import you will 
see a lot of processing statistics printed on the screen.  If the Import 
completed successfully you will see a blue "Import Complete" message.  
Everything is now set up and you can begin using Genmod with your 
GEDCOM.

*H.  Deleting GEDCOMs
You may delete GEDCOMs from your site from the "Manage GEDCOMs" area.
Deleting a GEDCOM from the site will remove it from the database but will 
not delete the original GEDCOM file that you imported.  It will also not
delete any of the cache or privacy files related to this GEDCOM.  These 
retained files, which are no longer required by Genmod, are all located 
in the "index" directory.

*I.  Reset config.php Permissions
For security you should set the permissions of config.php back to Read-only
(chmod 755) when you have finished configuring for the first time.  Write
permissions will only need to be set for config.php when you use the
Admin->Configuration link.  Everything else will be stored in the index
directory.

*J.  Custom Themes
You can customize the look and feel of Genmod by modifying one of the
provided themes.  See the THEMES section of this readme file for more
information.

*K.  HTTP Compression
Pages generated by Genmod can be large and use up a lot of bandwidth.
Compression of the data between the server and browser using GZip 
compression can compress the bandwidth by up to 90% (usually 80% - 90% for 
Genmod that were tested) If your web server is Apache, configuring this 
is very easy.  Add the following 2 lines to your php.ini file:
    php_flag zlib.output_compression On
    php_value zlib.output_compression_level 5

If you have no access to the php.ini file, create a blank file named 
.htaccess (including the dot) and add the lines to that file (or add them 
to an existing .htaccess file and upload the file to your Genmod 
directory.

Note: If your host is using mod_gzip or an other compression method, using 
this technique can cause problems.  Compression will have no effect on 
browsers that do not support it.  You can test the compression at
http://leknor.com/code/gziped.php

If you need help or support visit  http://www.genmod.net/

-------------------------------------------------------
UPDATING GEDCOMS

When you change your genealogy data outside of Genmod, it is not 

changelog.txt
=====================================================================
	Change Log
	Genmod

	This and other information can be found online at
	http://www.genmod.net/

	# $Id: changelog.txt,v 1.484 2009/03/29 13:20:10 sjouke Exp $
=====================================================================
Version 1.7 beta 1
- Added general search in media, notes and repositories (sjouke)
- Added tabbed results in search (sjouke)
- Added merge note records (sjouke)
- Fixed slow source list if link privacy is on (sjouke)
- Fixed show changes mechanism (sjouke)
- Added link tabs to all detail pages (sjouke)
- Added full support for level 0 notes (sjouke)
- Fixed Search: radio button show all/found names doesn't work (sjouke)
- Added display of media attached to notes in facts (sjouke)
- Fixed manage media layout issue on long filename (sjouke)
- Added option to build all unknown isdead statusses (sjouke)
- Added automatic install of languages in use from edit config page (sjouke)
- Added support for install/run Genmod from other ports than 80 (sjouke)
- Added general performance improvement by moving names to 1 table (sjouke)
- Added major performance improvement for soundex search (sjouke)
- Added query execution time to exec statistics (sjouke)
- Added souce/repo links in edit: click link for last added/selected (sjouke)
- Added support for BURI Y and CREM Y (sjouke)
- Added enhanced browser language detection (sjouke)
- Added moved all language variables to the database (sjouke)
- Fixed loading of languages during install (sjouke)
- Fixed edit gedcom configuration, gedcom language incorrect (sjouke)
- Fixed auto accept sometimes doesn't auto accept (sjouke)
- Added link in edit(add) to display next free ID (sjouke)
- Fixed update isdead status if dependent indi changes (sjouke)
- Fixed deadlock in indilist after show all (sjouke)
- Fixed chart boxes print double facts (sjouke)
- Added christening to add person form (sjouke)
- Fixed reorder children ignores christening date (sjouke)
- Fixed Cannot add note to asso on edit (sjouke)
- Fixed christening doesn't print in pedigree boxes (sjouke)
- Fixed two associates for one fact don't print properly (sjouke)
- Added indipage returns to last visited tab after refresh (sjouke)
- Fixed Feed doesn't display logo (sjouke)
- Added RSS block configuration (sjouke)
- Fixed default RSS format is not followed (sjouke)
- Fixed backslashes in find filter after search (sjouke)
- Added check for children order to sanity check (sjouke)
- Added basic support for level 0 note editing (sjouke)
- Fixed optimized memory usage of plot data calculation (sjouke)
- Fixed table keys too long on UTF-8 databases (sjouke)
- Added print error in install if table creation fails (sjouke)
- Fixed age of private associates is revealed (sjouke)
- Fixed export media DB->filesys creates empty thumbs (sjouke)
- Fixed medialist click on media shows empty pic (sjouke)
- Fixed error after jump between sites on same server (sjouke)
- Fixed cannot register new user if language switching is disabled (sjouke)
- Added edit option to immediately accept change if possible (sjouke)
- Fixed config problem in quickstart block (sjouke)
- Fixed problem in family-indi relation in DB (sjouke)
- Added # media objects to gedcom statistics block (sjouke)
- Fixed validity and root level domain check for email address (sjouke)
- Added block for top10 pageviews by bots (sjouke)
- Added split in hitcounts for users and bots (sjouke)
- Fixed error in search in changelog.php (sjouke)
- Fixed error in install in only english language is selected (sjouke)
- Added extended function to find primary family for pedigree (thomas)
- Fixed duplicate key error in reinstall (sjouke)
- Fixed double language string causes error in reinstall (sjouke)
- Fixed tiny privacy leak if 1 RESN is used (sjouke)
- Added most config variables moved to DB storage (sjouke)
- Fixed unnecessary renumbering of MM-items on import (sjouke)
- Fixed language strings are trimmed (sjouke)
- Fixed escaped placename shows after add person (sjouke)
- Added option to set the default page size for reports (sjouke)
- Fixed errors if famid is empty in family.php (sjouke)
- Added option to set delimiting characters for nicknames in names (sjouke)
- Added HTTP 304 handling to reduce spider traffic (sjouke, thomas)
- Added page specific meta tags (thomas)
- Fixed ID on individual page not removed if show id numbers is off (sjouke)
- Added changes to sources and repos to recent changes block (sjouke)
- Added FCKEditor also used by edit mypages (sjouke)
- Fixed imagview overlap of image and footer (sjouke)
- Added instructions to add FCKEditor to Genmod (thomas)
- Added statistics and statistics plot back to Genmod (thomas)
- Fixed alt text for delete file in media.php (sjouke)
- Fixed note records not recognised in merge gedcom (sjouke)
- Fixed error in export of clippings cart (sjouke)
- Fixed layout error in editconfig_gedcom (sjouke)
- Added check for equal first/lastname upon registration (sjouke)
- Fixed notes in multimedia references don't print on indipage main media (sjouke)
- Added GEDCOM creation time to the gedcom stats block (sjouke)
- Fixed problem with progress bars and compression settings by Thomas (sjouke)
- Fixed PHP5 error on indi page if viewed by visitor (sjouke)
- Fixed max execution time measurement must have max time to run (sjouke)
- Fixed no source link printed for facts if TITL and ABBR empty (sjouke)
- Fixed PHP error in merge gedcom if reference doesn't exist (sjouke)
- Fixed PHP error when deleting a note (sjouke)
- Added dead/alive determination also based on cremation/burial (sjouke)
- Fixed Genmod CVS version number to 1.7b1 (sjouke)
- Added user registration check for equal username/password (sjouke)
- Fixed edit, replace multimedia file adds multimedia dir to path (sjouke)

faq.php
<?php
/**
 * Customizable FAQ page
 *
 * Genmod: Genealogy Viewer
 * Copyright (C) 2005 - 2008 Genmod Development Team
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 *
 * This Page Is Valid XHTML 1.0 Transitional! > 01 September 2005
 *
 * @package Genmod
 * @subpackage Charts
 * @version $Id: faq.php,v 1.16 2008/08/03 04:29:47 sjouke Exp $
 */
 
/**
 * Inclusion of the configuration file
*/
require("config.php");

global $GM_IMAGES, $faqs;

if ($Users->userGedcomAdmin($gm_username)) $canconfig = true;
else $canconfig = false;
if (!isset($action)) $action = "show";
if (!isset($adminedit) && $canconfig) $adminedit = true;
else if (!isset($adminedit)) $adminedit = false;
$message = "";

// -- print html header information
print_header($gm_lang["faq_page"]);

if ($canconfig && $adminedit) {?>
	<!-- Setup the left box -->
	<div id="admin_genmod_left">
		<div class="admin_link"><a href="admin.php"><?php print $gm_lang["admin"];?></a></div>
	</div>
<?php } ?>
<div id="content">
<?php
	// NOTE: Commit the faq data to the DB
	if ($action=="commit") {
		if ($type == "update") {
			$faqs = GetFaqData();
			if (isset($faqs[$order])) {
				foreach ($faqs as $key => $item) {
					if ($key >= $order) {
						$sql = "UPDATE ".$TBLPREFIX."blocks SET b_order='".($key+1)."' WHERE b_id='".$faqs[$key]["header"]["pid"]."' and b_location='header'";;
						$res = NewQuery($sql);
						$sql = "UPDATE ".$TBLPREFIX."blocks SET b_order='".($key+1)."' WHERE b_id='".$faqs[$key]["body"]["pid"]."' and b_location='body'";
						$res = NewQuery($sql);
					}
				}
			}
			$sql = "UPDATE ".$TBLPREFIX."blocks SET b_order='".$order."', b_config='".$DBCONN->EscapeQuery(serialize($header))."' WHERE b_id='".$pidh."' and b_username='".$GEDCOM."' and b_location='header'";
			$res = NewQuery($sql);
			$sql = "UPDATE ".$TBLPREFIX."blocks SET b_order='".$order."', b_config='".$DBCONN->EscapeQuery(serialize($body))."' WHERE b_id='".$pidb."'  and b_username='".$GEDCOM."' and b_location='body'";
			$res = NewQuery($sql);
			WriteToLog("FAQ-> FAQ item has been edited.<br />Header ID: ".$pidh.".<br />Body ID: ".$pidb, "I", "G", $GEDCOM);
			$action = "show";
		}
		else if ($type == "delete") {
			$sql = "DELETE FROM ".$TBLPREFIX."blocks WHERE b_order='".$id."' AND b_name='faq' AND b_username='".$GEDCOM."'";
			$res = NewQuery($sql);
			WriteToLog("FAQ-> FAQ item has been deleted.<br />Header ID: ".$pidh.".<br />Body ID: ".$pidb, "I", "G", $GEDCOM);
			$action = "show";
		}
		else if ($type == "add") {
			$faqs = GetFaqData();
			if (isset($faqs[$order])) {
				foreach ($faqs as $key => $item) {
					if ($key >= $order) {
						$sql = "UPDATE ".$TBLPREFIX."blocks SET b_order='".($key+1)."' WHERE b_id='".$faqs[$key]["header"]["pid"]."' and b_location='header'";;
						$res = NewQuery($sql);
						$sql = "UPDATE ".$TBLPREFIX."blocks SET b_order='".($key+1)."' WHERE b_id='".$faqs[$key]["body"]["pid"]."' and b_location='body'";
						$res = NewQuery($sql);
					}
				}
			}
			$newid = GetNextId("blocks", "b_id");
			$sql = "INSERT INTO ".$TBLPREFIX."blocks VALUES ($newid, '".$GEDCOM."', 'header', '$order', 'faq', '".$DBCONN->EscapeQuery(serialize($header))."')";
			$res = NewQuery($sql);
			$sql = "INSERT INTO ".$TBLPREFIX."blocks VALUES (".($newid+1).", '".$GEDCOM."', 'body', '".$order."', 'faq', '".$DBCONN->EscapeQuery(serialize($body))."')";
			$res = NewQuery($sql);
			WriteToLog("FAQ-> FAQ item has been added.<br />Header ID: ".$newid.".<br />Body ID: ".($newid+1), "I", "G", $GEDCOM);
			$action = "show";
		}
		else if ($type == "moveup") {
			$faqs = GetFaqData();
			if ($id-1 != 0) {
				if (isset($faqs[$id-1])) {
					$sql = "UPDATE ".$TBLPREFIX."blocks SET b_order='".($id)."' WHERE b_id='".$faqs[$id-1]["header"]["pid"]."' and b_location='header'";;
					$res = NewQuery($sql);
					$sql = "UPDATE ".$TBLPREFIX."blocks SET b_order='".($id)."' WHERE b_id='".$faqs[$id-1]["body"]["pid"]."' and b_location='body'";
					$res = NewQuery($sql);
				}
				$sql = "UPDATE ".$TBLPREFIX."blocks SET b_order='".($id-1)."' WHERE b_id='".$pidh."' and b_location='header'";;
				$res = NewQuery($sql);
				$sql = "UPDATE ".$TBLPREFIX."blocks SET b_order='".($id-1)."' WHERE b_id='".$pidb."' and b_location='body'";
				$res = NewQuery($sql);
				WriteToLog("FAQ-> FAQ item has been moved up.<br />Header ID: ".$pidh.".<br />Body ID: ".$pidb, "I", "G", $GEDCOM);
			}
			else $message = "Item cannot be moved lower.";
			$action = "show";
		}
		else if ($type == "movedown") {
			$faqs = GetFaqData();
			if (isset($faqs[$id+1])) {
				$sql = "UPDATE ".$TBLPREFIX."blocks SET b_order='".($id)."' WHERE b_id='".$faqs[$id+1]["header"]["pid"]."' and b_location='header'";;
				$res = NewQuery($sql);
				$sql = "UPDATE ".$TBLPREFIX."blocks SET b_order='".($id)."' WHERE b_id='".$faqs[$id+1]["body"]["pid"]."' and b_location='body'";
				$res = NewQuery($sql);
			}
			
			$sql = "UPDATE ".$TBLPREFIX."blocks SET b_order='".($id+1)."' WHERE b_id='".$pidh."' and b_location='header'";;
			$res = NewQuery($sql);
			$sql = "UPDATE ".$TBLPREFIX."blocks SET b_order='".($id+1)."' WHERE b_id='".$pidb."' and b_location='body'";
			$res = NewQuery($sql);
			WriteToLog("FAQ item has been moved down.<br />Header ID: ".$pidh.".<br />Body ID: ".$pidb, "I", "G", $GEDCOM);
			$action = "show";
		}	
		$action = "show";
	}
	
	if ($action=="add") {
		$i=1;
		?>
		<form name="addfaq" method="post" action="faq.php">
			<input type="hidden" name="action" value="commit" />
			<input type="hidden" name="type" value="add" />
			<div class="admin_topbottombar">
				<?php print_help_link("add_faq_item_help","qm","add_faq_item"); ?>
				<?php print $gm_lang["add_faq_item"];?>
			</div>
			<div class="item_box shade2">
				<div class="width25 choice_left">
					<div class="helpicon"><?php print_help_link("add_faq_header_help","qm","add_faq_header"); ?></div>
					<div class="description"><?php print $gm_lang["add_faq_header"];?></div>
				</div>
			</div>
			<div class="item_box">
				<div class="width25 choice_left">
					<input type="text" name="header" size="90" tabindex="<?php print $i++; ?>" />
				</div>
			</div>
			<div class="item_box shade2">
				<div class="width25 choice_left">
					<div class="helpicon"><?php print_help_link("add_faq_body_help","qm","add_faq_body"); ?></div>
					<div class="description"><?php print $gm_lang["add_faq_body"];?></div>
				</div>
			</div>
			<div class="item_box">
				<div class="width25 choice_left">
					<textarea name="body" rows="10" cols="90" tabindex="<?php print $i++; ?>"></textarea>
				</div>
			</div>
			<div class="item_box shade2">
				<div class="width25 choice_left">
					<div class="helpicon"><?php print_help_link("add_faq_order_help","qm","add_faq_order"); ?></div>
					<div class="description"><?php print $gm_lang["add_faq_order"];?></div>
				</div>
			</div>
			<div class="item_box">
				<div class="width25 choice_left">
					<input type="text" name="order" size="3" tabindex="<?php print $i++; ?>" />
				</div>
			</div>
			<div class="item_box center">
				<input type="submit" name="submit" value="<?php print $gm_lang["save"]; ?>" tabindex="<?php print $i++;?>" />
			</div>
			<div class="admin_return"><a href="faq.php"><?php print $gm_lang["faq_back"];?></a></div>
		</form>
		<?php
	}
	
	if ($action == "edit") {
		if (!isset($id)) {
			$error = true;
			$error_message =  $gm_lang["no_id"];
			$action = "show";
		}
		else {
			$faqs = GetFaqData($id);
			$i=1;
			?>
			<form name="editfaq" method="post" action="faq.php">
				<input type="hidden" name="action" value="commit" />
				<input type="hidden" name="type" value="update" />
				<input type="hidden" name="id" value="<?php print $id;?>" />
				<div class="admin_topbottombar">
					<?php print_help_link("edit_faq_item_help","qm","edit_faq_item"); ?>
					<?php print $gm_lang["edit_faq_item"];?>
				</div>
				<?php foreach ($faqs as $id => $data) { ?>
					<input type="hidden" name="pidh" value="<?php print $data["header"]["pid"];?>" />
					<input type="hidden" name="pidb" value="<?php print $data["body"]["pid"];?>" />
					<div class="item_box shade2">
						<div class="width25 choice_left">
							<div class="helpicon"><?php print_help_link("add_faq_header_help","qm","add_faq_header"); ?></div>
							<div class="description"><?php print $gm_lang["add_faq_header"];?></div>
						</div>
					</div>
					<div class="item_box">
						<div class="width25 choice_left">
							<input type="text" name="header" size="90" value="<?php print $data["header"]["text"];?>" tabindex="<?php print $i++; ?>" />
						</div>
					</div>
					<div class="item_box shade2">
						<div class="width25 choice_left">
							<div class="helpicon"><?php print_help_link("add_faq_body_help","qm","add_faq_body"); ?></div>
							<div class="description"><?php print $gm_lang["add_faq_body"];?></div>
						</div>
					</div>
					<div class="item_box">
						<div class="width25 choice_left">
							<textarea name="body" rows="10" cols="90" tabindex="<?php print $i++; ?>"><?php print html_entity_decode(stripslashes($data["body"]["text"]));?></textarea>
						</div>
					</div>
					<div class="item_box shade2">
						<div class="width25 choice_left">
							<div class="helpicon"><?php print_help_link("add_faq_order_help","qm","add_faq_order"); ?></div>
							<div class="description"><?php print $gm_lang["add_faq_order"];?></div>
						</div>
					</div>
					<div class="item_box">
						<div class="width25 choice_left">
							<input type="text" name="order" size="3" value="<?php print $id;?>" tabindex="<?php print $i++; ?>" />
						</div>
					</div>
				<?php } ?>
				<div class="item_box center">
					<input type="submit" name="submit" value="<?php print $gm_lang["save"]; ?>" tabindex="<?php print $i++;?>" />
				</div>
				<div class="admin_return"><a href="faq.php"><?php print $gm_lang["faq_back"];?></a></div>
			</form>
		<?php
		}
	}
	
	if ($action == "show") {
		$faqs = GetFaqData();
		?>
		<div class="admin_topbottombar">
			<?php print "<h3>".$gm_lang["faq_page"]."</h3>"; ?>
		</div>
			<?php
			if (count($faqs) == 0 && $canconfig) { ?>
				<div class="shade2 item_box">
					<?php print_help_link("add_faq_item_help","qm","add_faq_item"); ?>
					<a href="faq.php?action=add"><?php print $gm_lang["add_faq_item"];?></a>
				</div>
				<?php
			}
			else if (count($faqs) == 0 && !$canconfig) { ?>
				<div class="shade2 item_box">
					<div class="error"><?php print $gm_lang["no_faq_items"]; ?></div>
				</div>
				<?php
			}
			else { ?>
				<div class="shade2 item_box">
				<?php
				// NOTE: Add and preview link
				if ($canconfig && $adminedit) { ?>
					<div class="choice_left">
						<?php print_help_link("add_faq_item_help","qm","add_faq_item");?>
						<a href="faq.php?action=add"><?php print $gm_lang["add"];?></a>
					</div>
					<?php
				}
				
				if ($canconfig && $adminedit) { ?>
					<div class="choice_middle">
						<?php print_help_link("preview_faq_item_help","qm","preview_faq_item");?>
						<a href="faq.php?adminedit=0"><?php print $gm_lang["preview"];?></a>
					</div>
					<?php
				}
				else if ($canconfig && !$adminedit) {
					print_help_link("restore_faq_edits_help","qm","restore_faq_edits");
					print "<a href=\"faq.php?adminedit=1\">".$gm_lang["edit"]."</a>";
				}
				
				if ($canconfig && $adminedit) {
					if (isset($error)) print "<div class=\"topbottombar red\">".$error_message."</div>";
				}
				?>
				</div>
				<?php
				foreach($faqs as $id => $data) {
					if ($data["header"] && $data["body"]) { ?>
						<div class="item_box">
							<?php
							// NOTE: Print the position of the current item
							if ($canconfig && $adminedit) { ?>
								<div class="shade2 width30 faq_item_left">
									<div class="choice_left">
										<?php print $gm_lang["position_item"];?>: <?php print $id;?>
									</div>
								</div>
								<?php
							}
							// NOTE: Print the header of the current item
							?>
							
								<?php if ($canconfig && $adminedit) {?> 
									<div class="width70 faq_item_right">
										<div class="choice_right">
								<?php }
								else { ?>
									<div class="admin_topbottombar">
								<?php } ?>
									<?php print html_entity_decode($data["header"]["text"]);?>
								<?php if ($canconfig && $adminedit) { ?> 
										</div>
									</div>
								<?php }
								else { ?>
									</div>
								<?php } ?>
						</div>
						<div class="item_box">
							<div class="shade2 width30 faq_item_left">
								<?php
								// NOTE: Print the edit options op the current item
								if ($canconfig && $adminedit) { ?>
									<div class="choice_left">
										<?php print_help_link("moveup_faq_item_help","qm","moveup_faq_item");?>
										<a href="faq.php?action=commit&amp;type=moveup&amp;id=<?php print $id; ?>&amp;pidh=<?php print $data["header"]["pid"]; ?>&amp;pidb=<?php print $data["body"]["pid"];?>"><img src="<?php print $GM_IMAGE_DIR."/".$GM_IMAGES["uarrow"]["other"];?>" alt="" /></a>
									</div>
									<div class="choice_middle">
										<?php print_help_link("movedown_faq_item_help","qm","movedown_faq_item"); ?>
										<a href="faq.php?action=commit&amp;type=movedown&amp;id=<?php print $id; ?>&amp;pidh=<?php print $data["header"]["pid"]; ?>&amp;pidb=<?php print $data["body"]["pid"];?>"><img src="<?php print $GM_IMAGE_DIR."/".$GM_IMAGES["darrow"]["other"];?>" alt="" /></a>
									</div>
									<div class="choice_middle">					
										<?php print_help_link("edit_faq_item_help","qm","edit_faq_item"); ?>
										<a href="faq.php?action=edit&amp;id=<?php print $id;?>"><?php print $gm_lang["edit"];?></a>
									</div>
									<div class="choice_middle">
										<?php print_help_link("delete_faq_item_help","qm","delete_faq_item"); ?>
										<a href="faq.php?action=commit&amp;type=delete&amp;id=<?php print $id;?>&amp;pidh=<?php print $data["header"]["pid"]; ?>&amp;pidb=<?php print $data["body"]["pid"];?>" onclick="return confirm('<?php print $gm_lang["confirm_faq_delete"];?>');"><?php print $gm_lang["delete"];?></a>
									</div>
									<?php
								}
								// NOTE: Print the body text op the current item
								?>
							</div>
							<?php
							if ($canconfig && $adminedit) {?> <div class="width70 faq_item_right"> <?php } ?>
								<div class="choice_right"><?php print nl2br(html_entity_decode(stripslashes($data["body"]["text"])));?></div>
							<?php if ($canconfig && $adminedit) { ?> </div> <?php } ?>
						</div>
						<?php
					}
				}
			}
	}
	if ($action != "show") {
		?>
		<script language="JavaScript" type="text/javascript">
		<!--
			document.<?php print $action;?>faq.header.focus();
		//-->
		</script>
		<?php
	}
	if ($canconfig && $adminedit) {
		if ($message != "") print "<div class=\"shade1\">".$message."</div>"; ?>
	<?php } ?>
</div>

<?php
print_footer();
?>
